neutronics/parse_outputs.py

```diff
@@ -0,0 +1,198 @@
+import operator
+import math
+import numpy as np
+import glob
+import sys
+
+# module imports
+import neutronic_sweeps as ns
+import material_data as md
+
+# sampling parameters from neutronic_sweeps
+sampled_params = ns._dimensions
+res = ['keff', 'mass', 'q_dens', 'BOL_U', 'EOL_U', 'rel_depl', 'dU']
+names = sampled_params + res
+types = ['f8']*len(names)
+
+# conversion factors
+g_to_kg = 1e-3
+cc_to_l = 1e-3
+kW_to_W = 1e3
+
+def load_outputs(data_dir):
+    """Load the MCNP output file string.
+
+    Arguments:
+    ----------
+        data_dir (str): path to dir containing MCNP6 output files
+    Returns:
+    --------
+        file_strings (list): list of lists of strings, 
+    1 list for each output file.
+    """
+    file_strings = []
+    files = glob.glob(data_dir)
+    for output in files:
+        # load file and read the lines
+        fp = open(output, 'r')
+        file_strings.append(fp.readlines())
+    
+    return file_strings
+
+def parse_header_string(lines, data):
+    """Get reactor parameters from header string of MCNP6 output file
+    
+    Arguments:
+    ----------
+        lines (list): list of strings (one for each line)
+        data (ndarray): container to store results
+    """
+    for line in lines:
+        if '1-' in line:
+            vals = [float(x) for x in line.split()[1].split(',')[:6]]
+            for key, val in zip(ns._dimensions, vals):
+                data[key] = val
+            # terminate parsing when header found
+            break
+
+def parse_keff(lines, data):
+    """Parse the EOL keff from the output file.
+    
+    Arguments:
+    ----------
+        lines (list): lines of the output file (to be parsed)
+        data (ndarray): container to store results
+    """
+    res_loc = []
+    
+    depletion_table_offset = 8
+    for idx, line in enumerate(lines):
+        if 'final result' in line:
+            res_loc.append(idx)
+
+    data['keff'] = float(lines[res_loc[-1]].split()[2])
+
+def calc_fuel_mass_q_dens(data):
+    """Calculate the reactor fuel mass and coolant density.
+    
+    Arguments:
+    ----------
+        data (ndarray): container to store results
+    """
+    core_r, r, PD, Q = data[['core_r', 'cool_r', 'PD', 'power']]
+    # clad thickness
```

Comment:
better to use a clearer variable name than to add a comment to describe your variable name

Reply:
I used c because it's a common variable used for clad thickness. I can change it though.